toolchain/check/facet_type.cpp

```diff
@@ -527,6 +535,18 @@ class SubstImplWitnessAccessCallbacks : public SubstInstCallbacks {
       }
     }
 
+    // If the access is going through a nested `ImplWitnessAccess`, that
+    // access needs to be resolved to a facet value first. If it can't be
+    // resolved then the outer one can not be either.
+    if (auto lookup = context().insts().TryGetAs<SemIR::LookupImplWitness>(
+            rhs_access->witness_id)) {
+      if (context().insts().Is<SemIR::ImplWitnessAccess>(
+              lookup->query_self_inst_id)) {
+        substs_in_progress_.push_back(rhs_inst_id);
+        return SubstResult::SubstOperandsAndAgain;
+      }
+    }
```

Comment:
What guarantee do we have that we'll make progress here? If substituting the operands doesn't change anything, can we end up back here with the same `ImplWitnessAccess` and loop infinitely? (If this can't happen, it'd be useful to have a comment here explaining why not.)